Central-Server/API/main.py

```diff
@@ -1,38 +1,151 @@
+Michael Karpov (2027)
+Zyun Lam (2027)
+Surag Nuthulapaty (2027)
+Carson Sprague (2025)
+Deeya Bodas (2024)
+Anthony Smykalov (?)   
+
+Under oversight of   
+Aidan Costello (2026)
+Peter Giannetos (2025)
+
+------------------------------------------------------------------------------
+
+Kamaji was developed with the express goal of making avionics development and testing a more streamlined process across Spaceshot.
+The documentation for this specific service is held within this directory's README.md file. This entrypoint file can be run as a standalone
+with { python ./main.py [dev/prod] }, or with the entire service by running the compose.prod.yml or compose.yml files as docker-compose instructions
+in the root of this repository.
+"""
+
+import sys
+import os
+import random
+import argparse
+
+from flask import Flask, jsonify, Response
+from flask_socketio import SocketIO
+from waitress import serve
+from blueprints.jobs_blueprint import jobs_blueprint
+from job_queue import job_queue_blueprint
+from blueprints.perms import perms_blueprint
+from apiflask import APIFlask
+from flask_cors import CORS
+
+import internal.boards
+import internal.database as database
+from internal.threads import BoardManagerThread
+
+# Handle main.py command line arguments
+parser = argparse.ArgumentParser()
+parser.add_argument("runmode", help="Enter the mode you want to run in: dev or prod", type=str)
+args = parser.parse_args()
+
+# Initialize the API
+app = APIFlask(__name__, title="Kamaji", static_url_path="/static", static_folder="./static")
+app.register_blueprint(jobs_blueprint)
+app.register_blueprint(job_queue_blueprint)
+app.register_blueprint(perms_blueprint)
+CORS(app)
+
+# Initialize datastreamer connectiosn
+m_thread = BoardManagerThread()
+
+# Set up server configs
+app.config['SPEC_FORMAT'] = 'json'
+app.config['SERVERS'] = [
+    {
+        'name': 'Dev Server',
+        'url': 'http://localhost/api/'
+    }
+    # TODO: Add prod
+]
+
+# Set up API endpoints
+@app.route("/")
+def api_index():
+    """/api/ -- Returns literal 'OK'"""
+    return "OK"
+
+@app.errorhandler(403)
+def access_forbidden_error(e):
+    """err 403 -- Returns normal 403 error"""
+    if random.randint(0, 100000) == 42069: # We are a really serious space program, check out the CEO of SpaceX, we like the same jokes
+        return "<html><h1>Access forbidden :(</h1><img src=\"/api/static/image_480.png\"/></html>", 200
+    return jsonify(error=str(e)), 403
+
+def generate_jobs_table():
+    """Generates database if it doesn't exist"""
+    conn = database.connect()
+    cursor = conn.cursor()
+    cursor.execute("SELECT 1 FROM pg_tables WHERE tablename='hilsim_runs'")
+    exists = cursor.fetchone()
+    if not exists:
+        with open("database.sql") as f:
+            cursor.execute(f.read())
+        conn.commit()
+        conn.close()
+        cursor.close()
+
+@app.route('/boards', methods=["GET"])

```

Comment:
This should be placed in a different file, but it needs access to the thread manager variable. Should we make the thread manager variable global?

Reply:
Let's talk about implementation at our next meeting.